pipert/utils/logger_utils.py

```diff
@@ -0,0 +1,48 @@
+import logging
+import os
+from logging.handlers import SocketHandler, TimedRotatingFileHandler
+
+
+def create_parent_logger(name):
+    logger_type = os.environ.get('LOGGER', 'file').lower()
+    return getattr(LoggerGenerator, f"_create_{logger_type}_logger")(name)
+
+
+class LoggerGenerator:
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def _create_cutelog_logger(name):
+        logger = logging.getLogger(name)
+        logger.setLevel(logging.DEBUG)
+        logger.propagate = False
+        socket_handler = SocketHandler('127.0.0.1', 19996)
+        socket_handler.setFormatter(logging.Formatter(
+            "%(asctime)s - %(levelname)s - %(name)s - %(message)s"))
+        logger.addHandler(socket_handler)
+        return logger
+
+    @staticmethod
+    def _create_file_logger(name):
+        class FileLogger(logging.Logger):
```

Comment:
Nested classes are fine but why would you define a nested class within a static function? It makes no real difference but I doubt that it's good practice to define a class within a function of any kind.

Reply:
I saw no reason to define it outside of the function because no one will ever use it.
I didn't want the code to be messy so i put it in the function (which is the only one using it).